### PR TITLE
Add School Document Model

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Notion ticket link

[Create schools model](https://www.notion.so/uwblueprintexecs/Create-schools-model-af7ab0ebfb36428f95718480a9677649)

## Implementation description
* Created a new document model for the `School` collection

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/models/school.model.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * This document contains information about a single school.
+ */
+export interface School extends Document {
+  /** the unique identifier for the school */
+  id: string;
+  /** the name of the school */
+  name: string;
+  /** the country the school is located in */
+  country: string;
+  /** the city the school is located in */
```
Comment (carissa-tang) [inline backend/typescript/models/school.model.ts:13]: did we choose not to include `sub-region`? i see it in the schema

Comment (mahadzaryab1) [reply backend/typescript/models/school.model.ts:15]: good catch! I missed it - fixed now.

**Review by carissa-tang — APPROVED** [review]

File: backend/typescript/models/school.model.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  subRegion: {
+    type: String,
+    required: true,
+  },
+  address: {
+    type: String,
+    required: true,
+  },
+  teachers: {
+    type: [String],
```
Comment (joyce-shi) [inline backend/typescript/models/school.model.ts:45]: Is there a reason why we reference teachers using `String` instead of `ObjectId`?

**Review by joyce-shi — CHANGES_REQUESTED** [review]
Left a quick comment. Other than that looks good!

File: backend/typescript/models/school.model.ts (hunk 2 above)
Comment (mahadzaryab1) [reply backend/typescript/models/school.model.ts]: fixed

**Review by joyce-shi — APPROVED** [review]
LGTM! 🛳

**Pull request merged**